File: src/main/resources/static/app/js/service/environment-service.js
dockerApp.service('environmentService', function (restService) {

    this.getAllEnvironments = function () {
        return restService.get('/environment/all');
    };

    this.getEnvironment = function (id) {
        return restService.get('/environment/' + id);
    };

    this.create = function (environment) {
        return restService.post('/environment', environment);
    };

    this.addHost = function (environment, host) {
        return restService.post('/environment/' + environment.id, host);
    };

});

File: src/main/java/ru/tisov/denis/platform/service/EnvironmentService.java
package ru.tisov.denis.platform.service;

import ru.tisov.denis.platform.domain.Environment;
import ru.tisov.denis.platform.domain.Host;

import java.util.List;

public interface EnvironmentService {

    List<Environment> getAll();

    Environment getById(Long id);

    Environment create(Environment environment);

    Environment addHost(Long environmentId, Host host);
}
